lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -3,17 +3,204 @@
  * Licensed under the MIT License.
  */
 
+import {
+    AuthError,
+    ClientAuthErrorCodes,
+    createClientAuthError,
+    HttpStatus,
+    INetworkModule,
+    NetworkResponse,
+    NetworkRequestOptions,
+    Logger,
+    UrlString,
+    ResponseHandler,
+    ServerAuthorizationTokenResponse,
+} from "@azure/msal-common";
 import { ManagedIdentityId } from "../../config/ManagedIdentityId";
 import { ManagedIdentityRequestParameters } from "../../config/ManagedIdentityRequestParameters";
 import { BaseManagedIdentitySource } from "./BaseManagedIdentitySource";
+import { CryptoProvider } from "../../crypto/CryptoProvider";
+import {
+    ManagedIdentityErrorCodes,
+    createManagedIdentityError,
+} from "../../error/ManagedIdentityError";
+import {
+    AUTHORIZATION_HEADER_NAME,
+    HttpMethod,
+    METADATA_HEADER_NAME,
+    ManagedIdentityIdType,
+} from "../../utils/Constants";
+import { NodeStorage } from "../../cache/NodeStorage";
+import * as fs from "fs";
+import { ManagedIdentityTokenResponse } from "../../response/ManagedIdentityTokenResponse";
 
+const ARC_API_VERSION: string = "2019-11-01";
+
+/**
+ * Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/AzureArcManagedIdentitySource.cs
+ */
 export class AzureArc extends BaseManagedIdentitySource {
+    private endpoint: string;
+
+    constructor(
+        logger: Logger,
+        nodeStorage: NodeStorage,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider,
+        endpoint: string
+    ) {
+        super(logger, nodeStorage, networkClient, cryptoProvider);
+
+        this.endpoint = endpoint;
+    }
+
+    public static tryCreate(
+        logger: Logger,
+        nodeStorage: NodeStorage,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider
+    ): AzureArc | null {
+        const imdsEndpoint: string | undefined = process.env["IMDS_ENDPOINT"];
```

Comment:
nit: what is the best practice for `falsy` in MSAL JS? I find it odd that imdsEndpoint can be undefined. I would think it can be null, to denote absence of a value. 

https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Reply:
As an environment variable, it's either defined or it's not. I'm not sure what you're getting at.